### PR TITLE
add deployment config for gnosis wsteth and sxdai hyperdrives

### PR DESCRIPTION
**Pull request by mcclurejt**
# Resolved Issues

# Description

Contains configuration for the following items on Gnosis chain:
- Factory
- Delv Registry
- Chainlink Deployer Coordinator
- ERC4626 Deployer Coordinator
- wstETH Hyperdrive instance
- sxDAI Hyperdrive instance

# Review Checklists

Please check each item **before approving** the pull request. While going
through the checklist, it is recommended to leave comments on items that are
referenced in the checklist to make sure that they are reviewed. If there are
multiple reviewers, copy the checklists into sections titled `## [Reviewer Name]`.
If the PR doesn't touch Solidity, the corresponding checklist can
be removed.

## [[Reviewer Name]]

- [ ] **Tokens**
    - [ ] Do all `approve` calls use `forceApprove`?
    - [ ] Do all `transfer` calls use `safeTransfer`?
    - [ ] Do all `transferFrom` calls use `msg.sender` as the `from` address?
        - [ ] If not, is the function access restricted to prevent unauthorized
              token spend?
- [ ] **Low-level calls (`call`, `delegatecall`, `staticcall`, `transfer`, `send`)**
    - [ ] Is the returned `success` boolean checked to handle failed calls?
    - [ ] If using `delegatecall`, are there strict access controls on the
          addresses that can be called? It shouldn't be possible to `delegatecall`
          arbitrary addresses, so the list of possible targets should either be
          immutable or tightly controlled by an admin.
- [ ] **Reentrancy**
    - [ ] Are functions that make external calls or transfer ether marked as `nonReentrant`?
        - [ ] If not, is there documentation that explains why reentrancy is
              not a concern or how it's mitigated?
- [ ] **Gas Optimizations**
    - [ ] Is the logic as simple as possible?
    - [ ] Are the storage values that are used repeatedly cached in stack or
          memory variables?
    - [ ] If loops are used, are there guards in place to avoid out-of-gas
          issues?
- [ ] **Visibility**
    - [ ] Are all `payable` functions restricted to avoid stuck ether?
- [ ] **Math**
    - [ ] Is all of the arithmetic checked or guarded by if-statements that will
          catch underflows?
    - [ ] If `Safe` functions are altered, are potential underflows and
          overflows caught so that a failure flag can be thrown?
    - [ ] Are all of the rounding directions clearly documented?
- [ ] **Testing**
    - [ ] Are there new or updated unit or integration tests?
    - [ ] Do the tests cover the happy paths?
    - [ ] Do the tests cover the unhappy paths?
    - [ ] Are there an adequate number of fuzz tests to ensure that we are
          covering the full input space?

### CODE REVIEW

File: tasks/deploy/config/gnosis/factory.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            checkpointRewarder: zeroAddress,
+            checkpointDurationResolution: parseDuration("1 hours"),
+            minCheckpointDuration: parseDuration("24 hours"),
+            maxCheckpointDuration: parseDuration("24 hours"),
+            minPositionDuration: parseDuration("7 days"),
+            maxPositionDuration: parseDuration("730 days"),
+            minFixedAPR: parseEther("0.005"),
+            maxFixedAPR: parseEther("0.1"),
+            minTimeStretchAPR: parseEther("0.005"),
+            maxTimeStretchAPR: parseEther("0.1"),
```
Comment (jalextowle) [inline tasks/deploy/config/gnosis/factory.ts:45]: Let's bump this to 20%

File: tasks/deploy/config/gnosis/sxdai-182day.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        ]);
+        let pc = await hre.viem.getPublicClient();
+        await pc.waitForTransactionReceipt({ hash: tx });
+    },
+    poolDeployConfig: async (hre) => {
+        return {
+            baseToken: WXDAI_ADDRESS_GNOSIS,
+            vaultSharesToken: SXDAI_ADDRESS_GNOSIS,
+            circuitBreakerDelta: parseEther("0.05"),
+            minimumShareReserves: parseEther("10"),
```
Comment (jalextowle) [inline tasks/deploy/config/gnosis/sxdai-182day.ts:60]: This can be 1e15

File: tasks/deploy/config/gnosis/sxdai-182day.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            minimumTransactionAmount: parseEther("0.001"),
+            positionDuration: parseDuration(SIX_MONTHS),
+            checkpointDuration: parseDuration("1 day"),
+            timeStretch: 0n,
+            // TODO: Read from the factory.
```
Comment (jalextowle) [inline tasks/deploy/config/gnosis/sxdai-182day.ts:65]: When are you planning to update this? Just curious

File: tasks/deploy/config/gnosis/sxdai-182day.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            governance: (await hre.getNamedAccounts())["deployer"] as Address,
+            feeCollector: zeroAddress,
+            sweepCollector: zeroAddress,
+            checkpointRewarder: zeroAddress,
```
Comment (jalextowle) [inline tasks/deploy/config/gnosis/sxdai-182day.ts:69]: These should also be read from the factory while we're doing that.

File: tasks/deploy/config/gnosis/wsteth-182day.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            minimumShareReserves: parseEther("0.001"),
+            minimumTransactionAmount: parseEther("0.001"),
+            positionDuration: parseDuration(SIX_MONTHS),
+            checkpointDuration: parseDuration("1 day"),
+            timeStretch: 0n,
+            // TODO: Read from the factory.
+            governance: (await hre.getNamedAccounts())["deployer"] as Address,
+            feeCollector: zeroAddress,
+            sweepCollector: zeroAddress,
+            checkpointRewarder: zeroAddress,
```
Comment (jalextowle) [inline tasks/deploy/config/gnosis/wsteth-182day.ts:84]: Same here. If the plan is to do this now, let's not forget this config.

File: tasks/deploy/lib/constants.ts [hunk 6]
```diff
@@ -91,6 +91,32 @@ export const WSTETH_WHALE_MAINNET =
 export const SEPOLIA_USDE_ADDRESS =
     "0x9458CaACa74249AbBE9E964b3Ce155B98EC88EF2" as Address;
 
+// ╭─────────────────────────────────────────────────────────╮
+// │ Gnosis Addresses                                        │
+// ╰─────────────────────────────────────────────────────────╯
+
+export const CHAINLINK_AGGREGATOR_ADDRESS_GNOSIS =
+    "0x6dcF8CE1982Fc71E7128407c7c6Ce4B0C1722F55" as Address;
```
Comment (jalextowle) [inline tasks/deploy/lib/constants.ts:99]: Let's remove this. I have this constant in the test to make setting the storage variable easier, but we're never going to want to use this in prod.

**Review by jalextowle — APPROVED** [review]
It's looking good. I had a couple of comments, and it's g2g once those are addressed.

File: tasks/deploy/config/gnosis/sxdai-182day.ts (hunk 3 above)
Comment (mcclurejt) [reply tasks/deploy/config/gnosis/sxdai-182day.ts:65]: done :)

**Pull request merged**